### PR TITLE
add support for PASSWORD EXPIRE

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve? 
Add support for PASSWORD EXPIRE in ALTER USER statement. #409 

### What is changed and how it works?
MySQL syntax:
```
alter_user_stmt:
          alter_user_command alter_user_list require_clause
          connect_options opt_account_lock_password_expire_options
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by leoppro** [comment]
@tangenta PTAL

File: parser_test.go [hunk 1]
```diff
@@ -2513,6 +2513,10 @@ func (s *testParserSuite) TestPrivilege(c *C) {
 		{`ALTER USER 'root'@'localhost' IDENTIFIED BY 'new-password', 'root'@'127.0.0.1' IDENTIFIED BY PASSWORD 'hashstring'`, true, "ALTER USER `root`@`localhost` IDENTIFIED BY 'new-password', `root`@`127.0.0.1` IDENTIFIED BY PASSWORD 'hashstring'"},
 		{`ALTER USER USER() IDENTIFIED BY 'new-password'`, true, "ALTER USER USER() IDENTIFIED BY 'new-password'"},
 		{`ALTER USER IF EXISTS USER() IDENTIFIED BY 'new-password'`, true, "ALTER USER IF EXISTS USER() IDENTIFIED BY 'new-password'"},
+		{"alter user 'test@localhost' password expire;", true, "ALTER USER `test@localhost`@`%` PASSWORD EXPIRE"},
```
Comment (leoppro) [inline parser_test.go]: could you add some cases like `create user xxx password expire never` after line 2498

Comment (leiysky) [reply parser_test.go]: done

File: parser.y [hunk 2]
```diff
@@ -8455,11 +8455,12 @@ CreateRoleStmt:
 
 /* See http://dev.mysql.com/doc/refman/5.7/en/alter-user.html */
 AlterUserStmt:
-	"ALTER" "USER" IfExists UserSpecList
+	"ALTER" "USER" IfExists UserSpecList RequireClause ConnectionOptions PasswordOrLockOptions
 	{
 		$$ = &ast.AlterUserStmt{
 			IfExists: $3.(bool),
 			Specs: $4.([]*ast.UserSpec),
+			PasswordOrLockOptions: $7.([]*ast.PasswordOrLockOption),
```
Comment (leoppro) [inline parser.y]: add the following fields for AlterUserStmt please, and add test cases for it:
```
	TslOptions            []*TslOption
	ResourceOptions       []*ResourceOption
```

Comment (leoppro) [reply parser.y]: for `RequireClause`、`ConnectionOptions` and `PasswordOrLockOptions` stmts, TiDB is not supporting.
could you add warns for this stmts?

Comment (leiysky) [reply parser.y]: both create and alter stmts?

Comment (leoppro) [reply parser.y]: yes

Comment (leiysky) [reply parser.y]: PTAL

File: parser.y [hunk 3]
```diff
@@ -8455,12 +8460,20 @@ CreateRoleStmt:
...
 	{
-		$$ = &ast.AlterUserStmt{
+		stmt := &ast.AlterUserStmt{
 			IfExists: $3.(bool),
 			Specs: $4.([]*ast.UserSpec),
+			TslOptions: $5.([]*ast.TslOption),
+			ResourceOptions: $6.([]*ast.ResourceOption),
+			PasswordOrLockOptions: $7.([]*ast.PasswordOrLockOption),
 		}
+		if len(stmt.TslOptions) > 0 || len(stmt.ResourceOptions) > 0 || len(stmt.PasswordOrLockOptions) > 0 {
```
Comment (leoppro) [inline parser.y]: maybe it better that we throw a warning like this:
```
PasswordOrLockOptions:
	{
		l := []*ast.PasswordOrLockOption{}
		$$ = l
	}
|	PasswordOrLockOptionList
	{
		$$ = $1
// throw the warning
	}
```

File: parser_test.go [hunk 4]
```diff
@@ -2513,6 +2517,18 @@ func (s *testParserSuite) TestPrivilege(c *C) {
...
+		{"alter user 'test@localhost' password expire never;", true, "ALTER USER `test@localhost`@`%` PASSWORD EXPIRE NEVER"},
+		{"alter user 'test@localhost' password expire default;", true, "ALTER USER `test@localhost`@`%` PASSWORD EXPIRE DEFAULT"},
+		{"alter user 'test@localhost' password expire interval 3 day;", true, "ALTER USER `test@localhost`@`%` PASSWORD EXPIRE INTERVAL 3 DAY"},
+		{"ALTER USER 'ttt' REQUIRE X509;", true, "ALTER USER `ttt`@`%` REQUIRE X509"},
+		{"ALTER USER 'ttt' REQUIRE SSL;", true, "ALTER USER `ttt`@`%` REQUIRE SSL"},
```
Comment (leoppro) [inline parser_test.go]: could you add tests for this stmt please:
```
REQUIRE tls_option [AND] tls_option 
tls_option: {
   SSL
 | X509
 | CIPHER 'cipher'
 | ISSUER 'issuer'
 | SUBJECT 'subject'
}
```

Comment (leoppro) [reply parser_test.go]: both alter and create

File: parser.y [hunk 5]
```diff
@@ -8559,24 +8564,32 @@ RequireClause:
 			Type: ast.TslNone,
 		}
 		$$ = []*ast.TslOption{t}
+		yylex.AppendError(yylex.Errorf("TiDB does not support REQUIRE now, they would be parsed but ignored."))
```
Comment (leoppro) [inline parser.y]:
```
RequireClauseOpt:
{
// throw warning
}
|  RequireClause
{
$$ = $1
}
```
it's better i think

File: parser.y [hunk 6]
```diff
@@ -8622,6 +8635,8 @@ PasswordOrLockOptions:
 |	PasswordOrLockOptionList
 	{
 		$$ = $1
+		yylex.AppendError(yylex.Errorf("TiDB does not support PASSWORD now, they would be parsed but ignored."))
```
[the remark this answers is not in the document]
Comment (leoppro) [reply parser.y]: you need run `make parser`, rebuild `parser.go`

**Comment by leoppro** [comment]
LGTM

**Pull request merged**